kv/consul/client.go

```diff
@@ -78,6 +78,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet, prefix string) {
 	f.BoolVar(&cfg.ConsistentReads, prefix+"consul.consistent-reads", false, "Enable consistent reads to Consul.")
 	f.Float64Var(&cfg.WatchKeyRateLimit, prefix+"consul.watch-rate-limit", 1, "Rate limit when watching key or prefix in Consul, in requests per second. 0 disables the rate limit.")
 	f.IntVar(&cfg.WatchKeyBurstSize, prefix+"consul.watch-burst-size", 1, "Burst size used in rate limit. Values less than 1 are treated as 1.")
+	f.DurationVar(&cfg.CasRetryDelay, prefix+"consul.cas-retry-delay", 0, "Max duration to wait before retrying a Compare And Swap (CAS) operation.")
```

Comment:
```suggestion
	f.DurationVar(&cfg.CasRetryDelay, prefix+"consul.cas-retry-delay", 500*time.Milliseconds, "Duration to wait before retrying a Compare And Swap (CAS) operation.")
```

I think a reasonable non-zero default here would make sense, part of the current issue is we retry in a tight loop.

Also changed the comment a little because it's not a backoff, it's a fixed retry time.

Reply:
I just saw @pstibrany and I are going back and forth a little one this.

I would prefer not to include `Max` because it implies the delay could be a variable amount less than the specified value when the code currently does a fixed delay of exactly this amount.

Reply:
It's not backoff, but we use it as input into random function:

```
	sleepBeforeRetry := time.Duration(0)
	if c.cfg.CasRetryDelay > 0 {
		sleepBeforeRetry = time.Duration(rand.Int63n(c.cfg.CasRetryDelay.Nanoseconds()))
	}
```

So it's really a "maximum duration".

Reply:
Oh i'm sorry, read the code again, should have done this first, there is a `rand` in it

Reply:
Short non-zero default value makes sense to me. We should document it the change in the changelog.

Reply:
Given this, I believe @pstibrany wording is correct.

I also updated the default to 1s